server/controllers/auth/signup.js

```diff
@@ -1,3 +1,39 @@
-exports.signup = (req, res) => {
-  res.send(' Hello from signup file');
+const { hash } = require('bcrypt');
+const { jwtSign } = require('../../helpers');
+const { signupSchema } = require('./validation/signup-validation');
+const { users: { getUserByEmailOrUsername, addUser, addUserInterests } } = require('../../database/queries');
+
+
+exports.signup = (req, res, next) => {
+  const key = process.env.KEY;
+  const {
+    username, email, password, nativeLangId, learningLangId, interestsId,
+  } = req.body;
+  signupSchema.validate({
+    username, email, password, nativeLangId, learningLangId, interestsId,
+  })
+    .then(() => getUserByEmailOrUsername(email, username))
+    .then(({ rows }) => {
+      if (rows.length !== 0) {
+        if (rows[0].username === username) throw Error('username exists');
+        if (rows[0].email === email) throw Error('email exists');
+      }
+    })
+    .then(() => hash(password, 10))
+    .then((hashed) => addUser({
```

Comment:
then here you can use the same variable to the object with giving the password the hashed
like this 

```suggestion
    .then((hashed) => addUser({ ...userIfo, password: hashed }))
```